services/src/main/java/org/keycloak/vault/PlainTextVaultProvider.java

```diff
@@ -0,0 +1,84 @@
+package org.keycloak.vault;
+
+import org.jboss.logging.Logger;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.EnumSet;
+import java.util.Optional;
+import java.util.function.BiFunction;
+
+/**
+ * A text-based vault provider, which stores each secret in a separate file. The file name needs to match a
+ * vault secret id (or a key for short). A typical vault directory layout looks like this:
+ * <pre>
+ *     ${VAULT}/realma__key1 (contains secret for key 1)
+ *     ${VAULT}/realma__key2 (contains secret for key 2)
+ *     etc...
+ * </pre>
+ * Note, that each key needs is prefixed by realm name. This kind of layout is used by Kubernetes by default
+ * (when mounting a volume into the pod).
+ *
+ * See https://kubernetes.io/docs/concepts/configuration/secret/
+ * See https://github.com/keycloak/keycloak-community/blob/master/design/secure-credentials-store.md#plain-text-file-per-secret-kubernetes--openshift
+ *
+ * @author Sebastian Łaskawiec
+ */
+public class PlainTextVaultProvider implements VaultProvider {
+
+    private static final Logger logger = Logger.getLogger(MethodHandles.lookup().lookupClass());
+
+    private final Path vaultPath;
+    private final BiFunction<Path, String, Path> secretMappingFunction;
+
+    /**
+     * Creates a new {@link PlainTextVaultProvider}.
+     *
+     * @param path A path to a vault. Can not be null.
+     * @param realmName A realm name. Can not be null.
+     */
+    public PlainTextVaultProvider(@Nonnull Path path, @Nonnull String realmName) {
+        vaultPath = path;
+        secretMappingFunction = (vaultDirectory, secretId) -> vaultDirectory.resolve(realmName + "__" + secretId);
+        logger.debugf("PlainTextVaultProvider will operate in %s directory", vaultPath.toAbsolutePath().toString());
```

Comment:
`vaultPath.toAbsolutePath().toString()` would be evaluated always which is redundant if debug is not enabled.

`        if (logger.isDebugEnabled()) {`
```suggestion
            logger.debugf("PlainTextVaultProvider will operate in %s directory", vaultPath.toAbsolutePath());
```
        }

Reply:
Sure, let me fix that.